app/presenters/queries/content_item_presenter.rb

```diff
@@ -46,10 +51,25 @@ def present
         end.compact
       end
 
+      def link_set_for(content_item_hash)
+        LinkSet.find_by!(content_id: content_item_hash.fetch("content_id"))
+      end
+
     private
 
       attr_accessor :content_item_scope, :fields
 
+      def link_sets_by_content_id
+        @link_sets ||= begin
+          content_ids = content_item_scope.pluck(:content_id)
+          link_sets = LinkSet.where(content_id: content_ids).includes(:links)
```

Comment:
Based on what we saw, I'm not sure this endpoint would be usable without tuning its performance?

Happy for that to go into a separate ticket.